### ISSUE TITLE
docker ps -a show huge Amount of Docker Containers on Mesos/Marathon Slave server 

### ISSUE THREAD

**Issue by cruizpollino** [mesosphere/marathon#2347]
Good evening,

We've configured a marathon 0.10.0 and mesos 0.23.0 with 3 masters and 4 slaves ( 2 of them with short amount of mem and cpu), we're using as well docker 1.7

We're running a POC of these technologies and we're deploying 1 docker container of logstash, 1 docker container of kibana, 2 instances of mesos-dns and a docker container cluster of elasticsearch with two instances

We've found out that in one of the servers with small amount of resources, where some of the marathon instances can't run due to lack of resources, when we run a "docker ps -a" we got a list of  almost 100 elements.

Is this behavior the expected one? Since the environment performs a pre-check of the resources on the slaves previous to try to start the instance task on the slave, this situation shall not happen right?

Kind Regards

**Comment by gkleiman** [comment]
Mesos/Marathon don't do a pre-check before starting tasks. Marathon gets resource offers from Mesos and then based on the resource requirements specified in the app definition, it will schedule tasks that fit in those offers.

Could it be possible that the resources you defined in the app definition are not enough to start the tasks? In that case Marathon would schedule a task, the Mesos Slave would then try to start it, but it would fail, freeing up the resources. Once Marathon notices that the task failed to start and that there are enough resources for it, it will reschedule it, leading to another task failure.

**Comment by kolloch** [comment]
Also if you start tasks not using Mesos, the auto-detection of Mesos will not be accurate.

I will close this now. If you have more questions, please ask.
